src/alfred/main.py

```diff
@@ -312,3 +325,40 @@ def _pythonpath(directories: List[str] = None, append_project=True) -> None:
     new_pythonpath = ":".join(real_directories + _pythonpath)
     with lib.override_pythonpath(new_pythonpath):
         yield
+
+
+def _parse_text_command(command: str) -> Tuple[LocalCommand, List[str]]:
+    """
+    Parse a text command and return a LocalCommand and a list of arguments
+
+    >>> command, args = _parse_text_command("echo hello world")
+    >>> # command == LocalCommand("echo")
+    >>> # args == ["hello", "world"]
+    >>>
+
+    This function does not handle shell operations like `|`, `>`, `>>`, etc...
+    These operations depend on the user's shell.
+
+    :param command:
+    :return:
+    """
+    cmd_parser = shlex.shlex(command, punctuation_chars=True)
+    cmd_parser.whitespace_split = True
+    cmd_parts = list(cmd_parser)
+    for index, part in enumerate(cmd_parts):
+        if part.startswith("'") and part.endswith("'"):
+            cmd_parts[index] = _parse_text_command_interpret_literal(part)
+        elif part.startswith('"') and part.endswith('"'):
+            cmd_parts[index] = _parse_text_command_interpret_literal(part)
+
+    contain_shell = any(True for part in cmd_parts if part in ['&', '|', '&&', '||', '>', '>>', '<', '<<'])
+    if contain_shell:
+        exception = click.ClickException(f"shell operations are not supported: `{command}`")
+        exception.exit_code = 1
+        raise exception
+    executable = sh(cmd_parts[0])
+    args = cmd_parts[1:]
+    return executable, args
+
+def _parse_text_command_interpret_literal(quoted_string: str):
+    return ast.literal_eval(quoted_string)
```

Comment:
I think this instruction is too permissive. Replace it with an unquote function which:

* eliminates external quotes 
* eliminates a first level of escape ``\"`` and ``\'``